lib/Dialect/FIRRTL/Transforms/LegacyWiring.cpp

```diff
@@ -0,0 +1,114 @@
+//===- LegacyWiring- legacy Wiring annotation resolver ----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the legacy Wiring annotation resolver.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/FIRRTL/AnnotationDetails.h"
+#include "circt/Dialect/FIRRTL/FIRRTLAnnotationHelper.h"
+#include "circt/Dialect/FIRRTL/FIRRTLAnnotations.h"
+#include "circt/Dialect/FIRRTL/FIRRTLUtils.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "mlir/Support/LogicalResult.h"
+
+using namespace circt;
+using namespace firrtl;
+
+/// Consume SourceAnnotation and SinkAnnotation, storing into state
+LogicalResult circt::firrtl::applyWiring(const AnnoPathValue &target,
+                                         DictionaryAttr anno,
+                                         ApplyState &state) {
```

Comment:
An issue with just converting legacy wiring problems into new wiring problems is that the new wiring problems will bore ref type ports along the source -> LCA path. The actual `WiringProblem` struct needs to be modified to include a bool indicating if it should create real or ref type ports and the `solveWiringProblem` logic needs to do the right thing here.